expression/util.go

```diff
@@ -415,7 +415,6 @@ func ColumnSubstituteImpl(expr Expression, schema *Schema, newExprs []Expression
 		if v.InOperand {
 			newExpr = SetExprColumnInOperand(newExpr)
 		}
-		newExpr.SetCoercibility(v.Coercibility())
```

Comment:
why remove this, rest is ok for me

Reply:
It may not be substituted in L424. If keep it, a test in constant propagation will fail.